File: sunken-webapp/src/main/java/io/sunken/Sunken.java
package io.sunken;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Verticle;
import org.slf4j.Logger;

import static io.sunken.HttpPath.httpPath;
import static io.vertx.ext.web.Router.router;
import static java.lang.System.currentTimeMillis;
import static org.slf4j.LoggerFactory.getLogger;

final class Sunken extends AbstractVerticle implements Verticle {


  private final HttpFork[] forks;

  Sunken(HttpFork[] forks) {
    this.forks = forks;
  }

  @Override
  public void start() {
  }

}

File: sunken-webapp/src/main/kotlin/io/sunken/Sunken.kt
package io.sunken

import io.sunken.HttpPath.httpPath
import io.vertx.core.AbstractVerticle
import io.vertx.core.Verticle
import io.vertx.ext.web.Router.router
import org.slf4j.LoggerFactory.getLogger
import java.lang.System.currentTimeMillis

class Sunken(vararg val forks: HttpFork) : AbstractVerticle(), Verticle {
  private val log = getLogger(Sunken::class.java)

  override fun start() {
    val startedMills = currentTimeMillis()
    vertx
      .createHttpServer()
      .requestHandler(
        httpPath(router(vertx), *forks).enter()
      )
      .listen(8080)
    log.info("Sunken has been started in {} ms.", currentTimeMillis() - startedMills)
  }
}

fun main(args: Array<String>) {
  Sunken().start()
}
